### PR TITLE
Allow user to specify a IPD if there's no Okta Session

### PR DESCRIPTION
**Pull request by FantasticFancy**
Signed-off-by: QiFanIress <Qi.Fan@iress.com>

### CODE REVIEW

File: build/build.js [hunk 1]
```diff
@@ -414,6 +414,9 @@ function genericOktaConfiguration() {
   config.AUTH_REQUEST.response_type = 'code';
   config.AUTH_REQUEST.scope = '${scope}';
   config.AUTH_REQUEST.redirect_uri = 'https://${domain-name}${callback-path}';
+  if ('${idp}') {
```
Comment (apgrucza) [inline build/build.js]: This line is redundant and should be removed. Note that this function runs at build time (not run time) and just generates a templated config file.

File: mocha/generic-config.json [hunk 2]
```diff
@@ -2,8 +2,9 @@
     "AUTH_REQUEST": {
         "client_id": "${client-id}",
         "response_type": "code",
-        "scope": "openid email",
-        "redirect_uri": "https://${domain-name}${callback-path}"
+        "scope": "${scope}",
+        "redirect_uri": "https://${domain-name}${callback-path}",
+        "idp": "${idp}"
```
Comment (apgrucza) [inline mocha/generic-config.json:7]: `idp` defaults to an empty string. Is it valid to pass an empty string for this parameter?

File: build/build.js (hunk 1 above)
Comment (FantasticFancy) [reply build/build.js]: Updated.

File: mocha/generic-config.json (hunk 2 above)
Comment (FantasticFancy) [reply mocha/generic-config.json:7]: Yes empty string is valid.

**Review by apgrucza — APPROVED** [review]

**Review by marwan-itani — APPROVED** [review]

**Pull request merged**